WordPress/Classes/ViewRelated/Post/AlertViewController+Autosave.swift

```diff
@@ -0,0 +1,54 @@
+import UIKit
+
+extension UIAlertController {
```

Comment:
It's great that you encapsulated the alert logic. I have a concern about using a `UIAlertController` extension though. 

This pollutes the `UIAlertController` namespace. The `static` method is exposed to everywhere even though they have no relation whatsoever to Posting/Editing. 

![image](https://user-images.githubusercontent.com/198826/69464889-e20e2c80-0d3c-11ea-9801-71d16bd4d7d7.png)

This gives the impression that the method is a “utility” even though it's not. It's only useable (or should only be used) in a single place. These consumers may not be able to use the method but I believe it still adds some cognitive overhead. 

My suggestion is to create a `PostListEditorPresenter` class that:

- Contains the [logic tree of whether to show the dialog or not](https://github.com/guarani/WordPress-iOS/blob/70e2da781cadb85a8494730d47a8cffc4a3ca276/WordPress/Classes/ViewRelated/Post/PostListViewController.swift#L500-L510)
- Presents the dialog if needed
- Presents the editor 

```swift 
    private func editPost(apost: AbstractPost) {
        guard let post = apost as? Post else {
            return
        }
        guard !PostCoordinator.shared.isUploading(post: post) else {
            presentAlertForPostBeingUploaded()
            return
        }

        let presenter = PostListEditorPresenter(presentingVC: self)
        presenter.startEditingFlow() // or something better :D 

    }
```

With this, we might even be able to make it testable. This class can be better. It might be trying to do too much. But I think the unit test can be valuable.

Reply:
Yeah, I agree this pollutes the `UIAlertController` namespace. I'll get to this tomorrow.

Reply:
Update now 👍